contributors/design-proposals/cluster-lifecycle/draft-20171020-bootstrap-checkpointing.md

```diff
@@ -0,0 +1,108 @@
+# Kubernetes Bootstrap Checkpointing Proposal
+
+## Metadata
+
+```yaml
+kep-number: draft-20171029-boostrap-checkpointing.md
+title: Kubernetes Bootstrap Checkpointing Proposal
+authors:
+  - name: Timothy St. Clair
+    github: @timothysc
+    email: timothysc@gmail.com
+owning-sig: sig-cluster-lifecycle 
+participating-sigs:
+  - sig-node
+reviewers:
+  - @yujuhong 
+  - @luxas 
+  - @roberthbailey
+approvers:
+  - @yujuhong
+  - @roberbailey
+editor:
+  name: @timothysc
+creation-date: 2017-10-20
+last-updated: 2017-10-20
+status: draft
+```
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Objectives](#objectives)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [User Stories](#user-stories)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Unresolved Questions](#unresolved-questions)
+
+## Summary
+
+There are several methods to deploy a kubernetes cluster, one method that offers some unique advantages is self hosting.  The purpose of this proposal is to outline a method to checkpoint specific annotated pods, namely the control plane components, for the purpose of enabling self hosting.  
+
+The details of self hosting are beyond the scope of this proposal, and are outlined in the references listed below: 
+
+  - [Self Hosted Kubernetes][0]
+  - [Kubeadm Self Hosting][1]
```

Comment:
this link points to the upgrades pull request, which doesn't seem correct.

Reply:
I'm about to write a dedicated self-hosting proposal soon-ish